src/shell.js

```diff
@@ -255,7 +255,7 @@ if (ENVIRONMENT_IS_NODE) {
   // not be needed with node v15 and about because it is now the default
   // behaviour:
   // See https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode
-  var nodeMajor = process.version.match(/^v(\d+)\./)[1];
+  var nodeMajor = Number(process.versions.node.split(".")[0]);
```

Comment:
Is the Number cast needed?  Its seems like the old code handles nodeMajor being a string.. and if you remove Number cast here then I think this can be code size saving?

Reply:
Yes - but it generally considered bad practice to do comparisons between numbers and strings. For reference this is how the nodejs codebase does version checks https://github.com/targos/node/blob/457b184f87c48a5f742c1caee3d6976fe8c01ad3/deps/undici/undici.js#L7804-L7805.

Reply:
I think in this case we would rather save a couple of bytes and go with the bad practice.. since the bad practice seems to work.

I suppose we could get bitten by some weird node version string one day.. but I think that would also cause the Number call to crash, so we could get bitten either way.

Its a bit like how we save a few byte by using `==` and `!=` in our `instanceof ` checks (#16261).  Its a bit naughty but we case about code size a lot and we code is auto-generated so doesn't really have a focus on human readability.